From 39ba2ca7dedd94269a5742241828e3886c990bcb Mon Sep 17 00:00:00 2001
From: Patrick Boog <jonesthefox@icloud.com>
Date: Fri, 30 Aug 2024 05:07:28 +0200
Subject: [PATCH] Give SD card higher priority than eMMC

In order to prevent the booting failure of OS in eMMC, keep having the SD card the higher priority than eMMC such that
the boot loader (U-Boot) in SD card, if exists, can start first and recover the OS installation. Therefore, this line
in the U-Boot source must be not changed [2]

[2] https://wiki.odroid.com/odroid-m1s/board_support/building_u-boot

---
 arch/arm/dts/rk3566-odroid-m1s.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/dts/rk3566-odroid-m1s.dts b/arch/arm/dts/rk3566-odroid-m1s.dts
index d7ceef1a1a7..6f462e10dcf 100644
--- a/arch/arm/dts/rk3566-odroid-m1s.dts
+++ b/arch/arm/dts/rk3566-odroid-m1s.dts
@@ -20,6 +20,7 @@
 
 	chosen {
 		stdout-path = "serial2:1500000n8";
+		u-boot,spl-boot-order = &sdmmc0, &sdhci;
 	};
 
 	hdmi-con {
-- 
2.39.2

